### ISSUE TITLE
I got 504 Gateway Time-out Error

### ISSUE THREAD

**Issue by takboo** [localtunnel/localtunnel#106]
Hi,I am sorry about this.i am a noob.
So today i first install localtunnel,then i run this command

``` bat
lt -p 8000.
```

i get this result.
![](http://i66.tinypic.com/2woydqt.png)
but when i vist this web.i got 504 error.
![](http://i68.tinypic.com/2d0m4r6.png)

what went wrong .please!

**Comment by davidthewatson** [comment]
I think this may be related to a change in OS X 10.11 El Capitan. 

I have identical web stacks running on two machines: one with Yosemite, the other with El Capitan. I just tried lt --port 5000 on both. The Yosemite machine works fine and the El Capitan machine doesn't get back to the web server running on port 5000. I've checked for firewall - none. 

I know there are other dev tool breakages on El Capitan so I'm skeptical. Can anyone confirm whether they've used lt successfully on OS X 10.11?

Thanks.

**Comment by iamfrontender** [comment]
Ubuntu 14.04, Latest update - same issue.

**Comment by Sheldan** [comment]
Fedora 21, same Problem

**Comment by aleclarson** [comment]
Not working for me on OS X 10.11.3 El Capitan

**Comment by lisavogtsf** [comment]
I seem to be having the same problem on OS X 10.11.4

**Comment by tlockAOB** [comment]
I have the same problem OSX Yosemite 10.10.5

**Comment by ehrig** [comment]
Same here on OS X 10.11.5 El Capitan

**Comment by Clam99** [comment]
I have successfully used it on 10.11.4 in the past.  However, I just went to use it again today (didn't do a software update or anything) and I am now getting the 504 error.

**Comment by matt-bailey** [comment]
Same for me, fresh install of El Capitan 10.11.6, fresh install of node/npm using homebrew, fresh install of localtunnel.

My local dev site is in a vagrant VM which I access locally using [my-domain].dev. It's on port 80 (I think, in that if I go to [my-domain].dev:80 I can still visit the site, but if I change that number I can't), and this is the port I used when setting up the tunnel: `lt --port 80`.

So I think I'm doing everything correctly?...

**Comment by fuzzy76** [comment]
Same here, is this just broken?

**Comment by devill** [comment]
This is still an issue! Does anyone has a workaround? It's supper annoying....

**Comment by iamfrontender** [comment]
You can try to use ngrok instead

**Comment by Clam99** [comment]
Yes, ngrock is the way to go. There doesn't seem to be a workaround for localtunnel.

**Comment by JerryGreen** [comment]
The issue is still present.

Another issue: https://github.com/localtunnel/localtunnel/issues/81

**Comment by Clam99** [comment]
Why did you close this, is it fixed?

**Comment by fuzzy76** [comment]
I am not the maintainer, but as far as I can tell there is a new release out that updates the request dependency which might (or might not) fix this.

**Comment by JerryGreen** [comment]
@Clam99 look at this https://github.com/localtunnel/localtunnel/issues/81#issuecomment-263136814

He is closing tasks. No guarantee it is fixed.

**Comment by hero0926** [comment]
It still happens on my win7 too

**Comment by Jakubscode** [comment]
Had same issue on Raspberry Pi with Linux. After some time I realized that i'm hosting app at my machine's ip "192.168.8.109:3000" not localhost.
so try `lt --port YOUR_PORT -local-host IP_ADDRESS_OF_YOUR_SERVER`
in my example  it was `lt --port 3000 -local-host 192.168.8.109`

**Comment by JerryGreen** [comment]
@Jakubscode do you mean this bug will no longer appear with such parameters?

Sounds a bit mad

**Comment by Jakubscode** [comment]
@JerryGreen 
I don't know why people had this issue, but in my case it was becuse of wrong ip adress.
Localtunnel works like this (simplified) :
- You have localtunnel server generally localtunnel.me, but also you can find repo on github and run your own
- localtunnel client, whitch is a node process on your local machine, that:
a) opens several (i.e 10) connection to localtunnel server
b) when localtunnel-server receives a request, it uses one of opened connections from client to pipe the request, then local client connect to your local app, get response, and returns it to server
Then server send you a response. 

If localtunnel client cannot connet to localtunnel-server you will receive an error, but when locatunnel client cannot connect to your app for some reason, you will receive an timeout. 
I'm not an expert in networking, but there is difference between ip `localhost` whitch is `127.0.0.1` and ip of your network card in local network, in my case  `192.168.8.109`. 
To sum up, just make sure that you are using correct ip adress.

**Comment by Deepakchawla** [comment]
same issue on Ubuntu 16.04... can anyone tell any solution of this..??

**Comment by arun2789** [comment]
Still experiencing this on Mac OS 10.12. Keeps timing out, and I've to re-initialize. Any fix timelines on this @defunctzombie?

**Comment by Nextt1** [comment]
I'm getting 504 - TIme-out error too. any solutions?

**Comment by AntonioCS** [comment]
I installed lt today and I keep getting connection errors over and over.
If the main suggestion is to use another project, would it not be better to just put this "advice" in the main page so people don't waste their time??
I understand this is open source etc, but if nobody is taking care of this project this should be made aware to the people that are trying to use it.
